src/Serialization/main.jl

```diff
@@ -183,7 +178,7 @@ end
 
 function load_internal(s::DeserializerState, ::Type{T}, dict::Dict) where T
```

Comment:
@lkastner @benlorenz So I discussed these generic (de)serialization methods with @fieker, and we think this is (too) dangerous. E.g. it can be used to (de)serialize `fmpq` -- except that this is fundamentally broken, and one may not notice this immediately, as it works fine for "small" fractions, but not beyond.

All in all, I don't think it's a good tradeoff to have a default that sometimes works and sometimes produces garbage. Better to not have a default and be forced to think about what one does.

So I think we must remove these -- or rather, I would suggest we rename them, say to `load_internal_generic`. And then we can selectively use it for types where it makes sense, via `load_internal(s,T::Type{MyCoolType}, dict) = load_internal_generic(s,T,dict)` etc.